warehouse/README.md

```diff
@@ -3,9 +3,60 @@
 This dbt project is intended to be the source of truth for the cal-itp-data-infra BigQuery warehouse.
 
 ##  Setting up the project on your local machine
-0. [Authorize the gcloud CLI](https://cloud.google.com/sdk/docs/authorizing) so that you can access BigQuery (implied step -1: make sure that you have permissions for BigQuery)
-1. Create a file `~/.dbt/profiles.yml` (i.e., in your home directory inside a `.dbt` subdirectory) that contains the following:
-```
+
+> Note: These instructions assume you are on macOS, but are largely similar for
+> other operating systems. Most *nix OSes will have a package manager that you
+> should use instead of Homebrew.
+
+> Note: if you get `Operation not permitted` when attempting to use the terminal,
+> you may need to [fix your terminal permissions](https://osxdaily.com/2018/10/09/fix-operation-not-permitted-terminal-error-macos/)
+
+### Install Homebrew (if you haven't)
+1. Follow the installation instructions at https://brew.sh/
+2. Then, `brew install gdal` to install a geospatial library needed later
+
+### Install the Google SDK (if you haven't)
+0. Implied: make sure that you have GCP permissions (i.e. that someone has added you to the GCP project)
+1. Follow the installation instructions at https://cloud.google.com/sdk/docs/install
+   1. If this is your first time using the terminal, we recommend reading https://blog.teamtreehouse.com/introduction-to-the-mac-os-x-command-line
+      or another tutorial first. You will generally need to understand `cd` and
+      the concept of the "home directory" aka `~`. When you first open the
+      terminal, your "working directory" will be the home directory. Running the
+      `cd` command without any arguments will set your working directory back to
+      `~`.
+   2. Use `tar -xvf <drag file from Finder window to get file name> ~` to unzip into your home directory
+   3. Answer `yes` to adding the tool to your path
+   4. Most recent macOS versions use `zsh` as the default shell, so ensure the path modification is added to `~/.zshrc` when prompted
+   5. Restart your terminal, and run `gcloud init`
+   6. Answer `yes` to log in, and select the Google account associated with GCP
+   7. Set `cal-itp-data-infra` as the default project, and do not set a region
+3. You should also [set the application default](https://cloud.google.com/sdk/gcloud/reference/auth/application-default)
+4. If `bq ls` shows output, you are good to go.
+
+
+### Install poetry and Python/dbt dependencies
+1. Install [poetry](https://python-poetry.org/docs/#osx--linux--bashonwindows-install-instructions) (used for package/dependency management).
+   1. If `curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -`
+      does not work, you can `curl` to a file `curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py`
```

Comment:
I think this just puts it on stdout -- should probably be `curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py -o get-poetry.py`